File: Pascal/nameTester.pas
program nameTester;
uses terminalUserInput;


procedure outputSillyName(yourName : String);
var
	i : Integer;

begin
	i := 0;
	Write (yourName, 'is a ');
	while (i<60) do
	begin
		Write ('silly ');
		i := i+1;
	end;
	Write('name!');
end;

procedure main();
var
	yourName : string;

begin
	yourName := readString('Please enter your name: ');

if yourName = 'Perrin' then
	WriteLn ('Awesome name!')
else
	outputSillyName(yourName);
end;


begin
	main();
end.

File: C/nameTester.c
#include <stdio.h>
#include <string.h>
#include "terminal_user_input.h"

#define LOOP_COUNT 60
#define GOOD_ONE "Jake"
#define GOOD_TWO "Tahnee"

int main()
{
    my_string name;
    int index;

    name = read_string("What is your name? ");

    if (strcmp(name.str, GOOD_ONE) == 0){
		printf(name.str);
        printf("%s, that's an awesome name!"); 
    }
    else if (strcmp(name.str, GOOD_TWO) == 0) {
        printf(name.str); 
        printf("%s, that's an awesome name!");
    }
    else {
        printf(name.str); 
        printf("%s is a ");
        for (index=0;index<LOOP_COUNT;index++) {
            printf("silly, ");
            }
        printf(" name");
    }
    return 0;
} 
 
    
